### ISSUE TITLE
tools.ps failure on Windows

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#1778]
```
==== resource:basis/tools/ps/ps-tests.factor

resource:basis/tools/ps/ps-tests.factor: 4

Unit Test: { { } [ ps. ] }

Win32 error 0x12b: Only part of a ReadProcessMemory or WriteProcessMemory request was completed.
(U) Quotation: [ c-to-factor => ]
    Word: c-to-factor
(U) Quotation: [ [ (get-catchstack) push ] dip call => (get-catchstack) pop* ]
(O) Word: command-line-startup
(O) Word: do-all
(O) Word: do-tests
(O) Word: test-vocabs
(O) Word: test-vocab
(O) Word: run-test-file
(U) Quotation: [
        2 load-locals -1 get-local <experiment> load-local
        0 get-local experiment. -2 get-local execute
        => 0 get-local -1 get-local [
            2 load-locals
            current-test-file get -1 get-local 0 get-local
            ~quotation~ curry curry ~quotation~ 2 drop-locals if
        ] curry curry [ drop ] 3 drop-locals if
    ]
    Word: experiment
(O) Word: (unit-test)
(U) Quotation: [
        [ ~quotation~ dip swap ~quotation~ dip ] dip swap
        [ call get-datastack ] dip => swap [ set-datastack ] dip
    ]
(U) Quotation: [ call => get-datastack ]
(O) Word: ps.
(O) Word: process-list
(O) Word: read-args
(O) Word: read-process-memory
(O) Word: (win32-error)
(O) Word: windows-error
(O) Method: M\ object throw
(U) Quotation: [
        OBJ-CURRENT-THREAD special-object error-thread set-global
        current-continuation => error-continuation set-global
        [ original-error set-global ] [ rethrow ] bi
    ]
```

**Comment by bpollack** [comment]
Did this just start, @mrjbq7? I don't have my Windows box quickly handy, but looking at the error and the relevant code makes it look like a struct misalignment. Is this on x86 or AMD64?

**Comment by mrjbq7** [comment]
On x86-64. It could be an race condition issue with reading memory from a process that closed at the same time?  Not an error I've seen before or frequently.

**Comment by erg** [comment]
Something like this should fix it assuming it won't infinite loop:

``tools.ps.windows``

```factor
:: read-process-memory ( HANDLE alien offset len -- byte-array )
    HANDLE
    offset alien <displaced-alien>
    len <byte-array> dup :> ba
    len
    f
    ReadProcessMemory 0x12b = [
        HANDLE alien offset len read-process-memory
    ] [
        win32-error=0/f
    ] if ba ;
```

**Comment by mrjbq7** [comment]
Fixed by 6ddeeac3dccb4ff3612e7b010308bae1b86ab02e.
